cms/server/contest/handlers/base.py

```diff
@@ -109,7 +116,7 @@ def write_error(self, status_code, **kwargs):
         # the data we need to display a basic template with the error
         # information. If r_params is not defined (i.e. something went
         # *really* bad) we simply return a basic textual error notice.
-        if getattr(self, 'r_params', None) is not None:
+        if self.r_params is not None:
```

Comment:
Revert or add `self.r_params = None` in `__init__`

Reply:
`self.r_params = None` is in the constructor of CommonRequestHandler